osu.Game/Database/DownloadableArchiveModelManager.cs

```diff
@@ -122,14 +113,27 @@ private void performDownloadWithRequest(ArchiveDownloadRequest<TModel> request)
                 {
                     request.Perform(api);
                 }
-                catch
+                catch (Exception e)
                 {
+                    // 404s (and maybe other failures) don't fire request.Failure so for now they handled here as well
```

Comment:
should be double-checked as to why before this is merged as a valid solution. (note to self if no one beats me).